### PR TITLE
change(chain): Drop transactions with unpaid actions

### LINKED ISSUE #8609: feature: Remove unpaid action limit
### Motivation

The [next zcashd release will remove the unpaid action limit](https://discord.com/channels/809218587167293450/809251012610752513/1250467188033458267), which was the workaround for wallets to keep working while they didn't support ZIP-317. We should do the same in zebra.

### Specifications

_No response_

### Complex Code or Requirements

blocks: #8321 

### Testing

_No response_

### Related Work

https://github.com/zcash/zcash/issues/6899

### PR DESCRIPTION
**Pull request by upbqdn**
## Motivation

Close #8609.

### Specifications & References

https://zips.z.cash/zip-0317

## Solution

- Rename `BLOCK_PRODUCTION_UNPAID_ACTION_LIMIT` to `BLOCK_UNPAID_ACTION_LIMIT`.
- Set `BLOCK_UNPAID_ACTION_LIMIT` to 0.
- Cosmetics & docs along the way.

### Tests

- Adjust existing tests.
- Add a new one to check if the mempool checks can pass with valid parameters.

### PR Author's Checklist

- [x] The PR name will make sense to users.
- [x] The PR provides a CHANGELOG summary.
- [x] The solution is tested.
- [x] The documentation is up to date.
- [x] The PR has a priority label.

### PR Reviewer's Checklist

- [ ] The PR Author's checklist is complete.
- [ ] The PR resolves the issue.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org).
   but only feature flags instead.
 - Fixed a bug with trailing characters in the openapi spec method descriptions ([#8597](https://github.com/ZcashFoundation/zebra/pull/8597))
 - Added default constructions for several RPC method responses([#8616](https://github.com/ZcashFoundation/zebra/pull/8616))
+- Zebra now drops transactions with unpaid actions.
```
Comment (arya2) [inline CHANGELOG.md]: Optional:

```suggestion
- Zebra now drops transactions with unpaid actions in block templates and from the mempool.
```

File: zebra-chain/src/transaction/unmined/zip317.rs [hunk 2]
```diff
@@ -198,6 +202,11 @@ pub fn mempool_checks(
     // In zcashd this is `DEFAULT_MIN_RELAY_TX_FEE` and `LEGACY_DEFAULT_FEE`:
     // <https://github.com/zcash/zcash/blob/f512291ff20098291442e83713de89bcddc07546/src/main.h#L71-L72>
     // <https://github.com/zcash/zcash/blob/9e856cfc5b81aa2607a16a23ff5584ea10014de6/src/amount.h#L35-L36>
+    //
+    // ## Note
+    //
+    // If the check above, which checks the number of unpaid actions, doesn't fail, then there is no
+    // way for the legacy check below to fail. This renders the legacy check redundant.
```
Comment (arya2) [inline zebra-chain/src/transaction/unmined/zip317.rs]: Optional/Nitpick: We could skip the next check and the last part of `select_mempool_transactions()` when producing block templates if the unpaid actions limit is 0.

```suggestion
    // If the check above for the maximum number of unpaid actions passes with the default 
    // unpaid action limit, then there is no way for the legacy check below to fail. 
    // This renders the legacy check redundant with the default unpaid action limit of 0.
```

**Review by arya2 — DISMISSED** [review]
This looks good! Thank you.

File: zebra-chain/src/transaction/unmined/zip317.rs (hunk 2 above)
Comment (upbqdn) [reply zebra-chain/src/transaction/unmined/zip317.rs]: I'll try to rephrase this again.

Comment (upbqdn) [reply zebra-chain/src/transaction/unmined/zip317.rs]: Done in [b266e40](https://github.com/ZcashFoundation/zebra/pull/8638/commits/b266e4052ad65533d9dc068c049bdf71294a9b06).

**Review by arya2 — DISMISSED** [review]

**Review by arya2 — APPROVED** [review]

**Comment by arya2** [comment]
Removed the `extra-reviews` label because it's not actually a consensus rule change, it only affects the transaction verifier tests for mempool requests. I reviewed it especially thoroughly since it is still sensitive code.

**Pull request merged**